### ISSUE TITLE
Question: How does the AMLEngine handle ambiguous paths in an AML file?

### ISSUE THREAD

**Issue by kk738** [AutomationML/AMLEngine2.1#22]
Is this valid AML having different AML Objects in this case an Attribute and RoleClass that are referenced by path (all but IEs and EIs) with the same name?

```XML
<RoleClassLib Name="MTPCommunicationRCLib">
    <Description>RoleClassLibrary for the Communcation Class</Description>
    <Version>0</Version>
    <RoleClass ID="1f317317-0e08-4176-abaa-64737fd8e973" Name="Timestamped" RefBaseClassPath="AutomationMLBaseRoleClassLib/AutomationMLBaseRole">
      <Description>Timestamp Extension for the last processing time of the DataAssembly</Description>
      <Attribute Name="Timestamp" AttributeDataType="xs:dateTime" />
      <RoleClass Name="Timestamp" />
    </RoleClass>
...
```

and how is this handled by the AMLEngine? Do I always get the first item then, or is it random?
Is this handled differently for IEs and EIs, as they are referenced by ID, so the same name of IE siblings should be allowed, right?

```C#
//test if fullnode path is the same --> yes they are
var el = caex.FindByID("1f317317-0e08-4176-abaa-64737fd8e973");
var path1 = (el.CAEXChild("Attribute") as ICAEXObject).GetFullNodePath();
var path2 = (el.CAEXChild("RoleClass") as ICAEXObject).GetFullNodePath();

//what item do I get here?
var ambiguousItem = caex.FindByPath("MTPCommunicationRCLib/Timestamped/Timestamp");
```

**Comment by josefprinz** [comment]
Attributes, assigned to classes should not be named equal to a subclass. The Path of the class attribute and the path of the subclass are equal and the engine cannot identify a unique item. Always the first identified item using the path is found which will be the attribute in this case, because the attribute precedes the subclass in the xml tree. 

I have noticed, that for these situations a new method will be helpful, which allows to identify a specific element type with a path.  I have added this to the new feature list for the nect update.
